gradle/wrapper/gradle-wrapper.properties

```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-all.zip
```

Comment:
Should it be `8.0-bin` to save traffic on CI builds?

```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
```

Reply:
It saves on CI builds, but seriously degrades local developer experience (code navigation). Especially that this is a Gradle plugin :)

Reply:
>It saves on CI builds, but seriously degrades local developer experience (code navigation)

Can you elaborate which code navigation it breaks?
IDEA resolves and downloads Gradle sources just fine even if you mention `-bin` distribution.

Reply:
Yeah, I've heard that a lot, but then I wasn't able to find/open classes from Gradle in a project which used -bin. I'll try to find an example.

Reply:
Here's what I did:
 * `8.0-all` (current branch state) -> navigation works
 * change `8.0-all` to `8.0-bin`
 * sync -> navigation still works,  
   but source code is loaded from `-all` folder in `wrapper/dists`
 * sync, restart, sync -> navigation still works,  
   but source code is loaded from `-all` folder in `wrapper/dists`
 * change `8.0-bin` to `8.0-rc-4-bin` to simulate a "fresh start"
 * sync -> I saw `gradle-8.0-rc-4-src.zip` (40M) downloaded -> navigation works
   this is when I started really testing

Here are some of the use cases that I tested apart from simple Ctrl+Click.

1. See all the files in a specific .jar file that contains the class:
   Works the better in `-all`.
   * navigate to class (e.g. `NamedDomainObjectContainer`)
   * press "Select Opened File" ("target" button) on Project view.
   * go to .jar's root and drill down to META-INF and other files/folders
   * when I go up to the .jar's root I see this:
     ![image](https://user-images.githubusercontent.com/2906988/219711668-28445d1b-ab5c-4ae3-8d35-3623cd9a7328.png)
   * At least in `-all` it shows the folder names:
     ![image](https://user-images.githubusercontent.com/2906988/219714748-565afaf7-3238-4d6d-98f3-9476cfabdd8f.png)
2. Navigate to a specific plugin by ID: Navigate > File ... > `org.gradle.java.properties`
   Works in both, except with `-bin` all the files are duplicated:
     ![image](https://user-images.githubusercontent.com/2906988/219712086-1dc3842d-bfd0-4ede-bcbb-3e6b9b8fb28c.png)
3. Focusing on any particular class doesn't show which .jar it belongs to  
   broken in both, this only works for normal dependencies
   * open some source code of any Gradle class
   * press "Select Opened File" ("target" button) on Project view.
   * Observe location
     * bin: ![image](https://user-images.githubusercontent.com/2906988/219713183-85b3ea08-60b1-42e4-9de7-9861c194ea2b.png)
     * all: ![image](https://user-images.githubusercontent.com/2906988/219715208-320bf1db-be41-4447-ad9b-50a4516d0c47.png)
4. Opening a .class file via navigate to file, e.g. `NamedDomainObjectContainer.class`
   (this happens sometimes automatically when debugging, probably more with Kotlin):
   broken in both, unless you explicitly attach the source code to the .class files.
    ```
    //
    // Source code recreated from a .class file by IntelliJ IDEA
    // (powered by FernFlower decompiler)
    //
    ```
    or even worse
    ```
    // IntelliJ API Decompiler stub source generated from a class file
    // Implementation of methods is not available
    
    package org.gradle.kotlin.dsl
    ... { /* compiled code */ }
    ```

So I'm not entirely sure what didn't work still. `-all` has a little bit better experience in advanced use cases, but generally looks the same, so [I'm lowering the load](https://github.com/gradle-nexus/publish-plugin/pull/168/commits/45afbcd498be398026dcb0a65e7fcae70d46c513); and I'll vigilantly watch out for WTFs related to Gradle source navigation.

Reply:
The duplicated `java` folder names look bizzare. It is probably worth creating a YouTrack issue for that. I've no idea how to search the one, so I would probably just file a new one.